pkg/workflow/flows.go

```diff
@@ -0,0 +1,418 @@
+package workflow
```

Comment:
I have read the code, but it involves some details. I think we need to have a discussion about it to help me have a better understanding of the code